### ISSUE TITLE
TestBody() function access error

### ISSUE THREAD

**Issue by anderalvp** [CUP-ECS/cajitafluids#52]
I'm getting a compilation error related with gtest when I execute the `make` command on Cajitfluids build directory:

> /home/anderson/cajitafluids/tests/tstProblemManager.cpp(45): error: The enclosing parent function ("TestBody") for an extended __host__ __device__ lambda cannot have private or protected access within its class

I tried changing the private access by public access on the TestBody function located at `/home/anderson/cajitafluids/cmake/blt/thirdparty_builtin/googletest-master-2020-01-07/googletest/include/gtest/internal/gtest-internal.h` but still the same error, any solution? I'm building this on a Tesla M10 GPU with CUDA support and this is my spack.yaml environment configuration:

![image](https://github.com/CUP-ECS/cajitafluids/assets/101995296/34e54016-bb71-4a29-bbcc-ca039d8785d2)

**Comment by patrickb314** [comment]
I'll take a closer look in about a week or so, but in the meantime the easiest things to do is probably to disable the tstProblemManager testcase in the tests/ CMakeFile. The test cases are not necessary for the program to run correctly.

**Comment by anderalvp** [comment]
Alright, I've disabled the tstProblemManager on tests/CMakeLists.txt and the compilation was successful, but when I execute `bin/advection -x cuda` to run the program I have the following issue:

![image](https://github.com/CUP-ECS/cajitafluids/assets/101995296/f1795da5-8c69-40c9-86d7-4647bf7de084)

This is because the CUDA ARCH? I'm using `cuda_arch=50` which is for GPU NVIDIA Tesla M10. I've executed the program on a generic multicore CPU without issues the first time.
